### PR TITLE
EnternewName state

### PR DESCRIPTION
**Pull request by humulla**
## Link to JIRA ticket

[jira Ticket](https://hackney.atlassian.net/browse/MTTL-2990)

## Describe this PR

Trigger EnterNewName State

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

**Review by notatiyyah — COMMENTED** [review]
I don't see any new E2E tests?

File: ProcessesApi.Tests/V1/E2ETests/Fixtures/ProcessFixture.cs [hunk 1]
```diff
@@ -82,17 +84,51 @@ private void createProcess(string state)
             });
         }
 
+        private void createConProcess(string state)
```
Comment (notatiyyah) [inline ProcessesApi.Tests/V1/E2ETests/Fixtures/ProcessFixture.cs:87]: Can't you just keep the createProcess function and have a param for process name?

File: ProcessesApi.Tests/V1/E2ETests/Fixtures/ProcessFixture.cs [hunk 2]
```diff
@@ -56,7 +58,7 @@ protected virtual void Dispose(bool disposing)
             }
         }
 
-        private void createProcess(string state)
+        private void createstjProcess(string state)
```
Comment (notatiyyah) [inline ProcessesApi.Tests/V1/E2ETests/Fixtures/ProcessFixture.cs:61]: A bit nit-picky but do you mind updating the casing? I read this as 'createst jProcess'

File: ProcessesApi.Tests/V1/E2ETests/Steps/UpdateChangeOfNameProcessSteps.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+namespace ProcessesApi.Tests.V1.E2ETests.Steps
+{
+    public class UpdateChangeOfNameProcessSteps : BaseSteps
+    {
+        private readonly IDynamoDbFixture _dbFixture;
```
Comment (notatiyyah) [inline ProcessesApi.Tests/V1/E2ETests/Steps/UpdateChangeOfNameProcessSteps.cs:26]: Is this similar to UpdateSTJProcessSteps? Maybe make a UpdateProcess base class with shared methods?

File: ProcessesApi.Tests/V1/Helpers/SharedProcessTests.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
+using FluentAssertions;
+using ProcessesApi.V1.Helpers;
+using ProcessesApi.V1.Services.Exceptions;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace ProcessesApi.Tests.V1.Helpers
+{
+    public class SharedProcessTests
```
Comment (notatiyyah) [inline ProcessesApi.Tests/V1/Helpers/SharedProcessTests.cs:10]: SharedProcessHelperTests?

File: ProcessesApi/V1/Helpers/ProcessHelper.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            });
+        }
+
+        public static void ValidateOptionalFormData(Dictionary<string, object> requestFormData, List<string> expectedFormDataKeys)
+        {
+            var atLeastOneKey = expectedFormDataKeys.Any(x => requestFormData.Any());
+            if (!atLeastOneKey)
+                throw new FormDataNotFoundException(requestFormData.Keys.ToList(), expectedFormDataKeys);
+
+            //expectedFormDataKeys.ForEach(x =>
```
Comment (notatiyyah) [inline ProcessesApi/V1/Helpers/ProcessHelper.cs:24]: Remove the comments if not needed?

File: ProcessesApi/V1/Helpers/ProcessHelper.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+            expectedFormDataKeys.ForEach(x =>
+            {
+                if (!requestFormData.ContainsKey(x))
+                    throw new FormDataNotFoundException(requestFormData.Keys.ToList(), expectedFormDataKeys);
+            });
+        }
+
+        public static void ValidateOptionalFormData(Dictionary<string, object> requestFormData, List<string> expectedFormDataKeys)
+        {
+            var atLeastOneKey = expectedFormDataKeys.Any(x => requestFormData.Any());
```
Comment (notatiyyah) [inline ProcessesApi/V1/Helpers/ProcessHelper.cs:20]: Does this work?

Comment (humulla) [reply ProcessesApi/V1/Helpers/ProcessHelper.cs:20]: Yep it does, I debugged and checked if there is no values given then it fails and if there is at least one then it passes

File: ProcessesApi.Tests/V1/Services/ChangeOfNameServiceTests.cs [hunk 7]
```diff
@@ -46,11 +47,40 @@ public async Task InitialiseStateToEnterNewNameIfCurrentStateIsNotDefinedAndTrig
...
+
+            // Assert
+            CurrentStateShouldContainCorrectData(process,
+                                                 triggerObject,
+                                                 ChangeOfNameStates.NameSubmitted,
+                                                 new List<string>() { /*TODO: Add next state here */ });
+            process.PreviousStates.LastOrDefault().State.Should().Be(ChangeOfNameStates.EnterNewName);
+            VerifyThatProcessUpdatedEventIsTriggered(ChangeOfNameStates.EnterNewName, ChangeOfNameStates.NameSubmitted);
+        }
+
```
Comment (notatiyyah) [inline ProcessesApi.Tests/V1/Services/ChangeOfNameServiceTests.cs:84]: Can you add a test that checks that it fails if none of the formdata fields are filled in?

File: ProcessesApi/V1/Helpers/ProcessHelper.cs (hunk 6 above)
Comment (notatiyyah) [reply ProcessesApi/V1/Helpers/ProcessHelper.cs:20]: Even if there is irrelevant form data, just not one of the required ones?
E.g.
```
{
  "some-field": "test"
}
```
Should still fail because it doesn't contain one of: title, firstName, LastName, etc

**Review by notatiyyah — APPROVED** [review]
